query/src/main/java/org/infinispan/query/dsl/embedded/impl/QueryEngine.java

```diff
@@ -706,12 +821,14 @@ protected IckleFilterAndConverter createFilter(String queryString, Map<String, O
          log.debugf("The resulting Lucene query is : %s", luceneQuery.toString());
       }
 
-      CacheQuery<?> cacheQuery = makeCacheQuery(ickleParsingResult, luceneQuery);
-      if (luceneParsingResult.getSort() != null) {
-         cacheQuery = cacheQuery.sort(luceneParsingResult.getSort());
-      }
-      if (luceneParsingResult.getProjections() != null) {
-         cacheQuery = cacheQuery.projection(luceneParsingResult.getProjections());
+      CacheQuery<?> cacheQuery = makeCacheQuery(ickleParsingResult, luceneQuery, queryMode, namedParameters);
+      if (queryMode != IndexedQueryMode.BROADCAST) {
```

Comment:
Why no sorting and projection in broadcast mode?

Reply:
Both are part of the query string in broadcast mode

Reply:
Seems like something I would write a short comment about, so I can remember later :)